pkg/hubble/exporter/exporter.go

```diff
@@ -132,17 +130,34 @@ func (e *exporter) Stop() error {
 	return err
 }
 
-// OnDecodedEvent checks if the event passes the filter.
-// If context was cancelled, it calls Stop() and stops processing events.
-func (e *exporter) OnDecodedEvent(_ context.Context, ev *v1.Event) (bool, error) {
+// OnDecodedEvent implements the observeroption.OnDecodedEvent interface.
+//
+// It takes care of applying filters on the received event, and if allowed, proceeds to invoke the
+// registered OnExportEvent hooks. If none of the hooks return true (abort signal) the event is then
+// wrapped in observerpb.ExportEvent before being json-encoded and written to its underlying writer.
+func (e *exporter) OnDecodedEvent(ctx context.Context, ev *v1.Event) (bool, error) {
 	select {
-	case <-e.ctx.Done():
-		return false, e.Stop()
+	case <-ctx.Done():
+		return false, nil
 	default:
 	}
```

Comment:
Nitpick: Now it's correct, but questionably useful. I don't think we need to handle the case where someone calls us with a canceled context. (But it's also fine to keep it)

Reply:
Its a good point.

I guess it depends if we want best-effort early exit to be handled by each hook (and let them be the judge) or at the top level call site where it could be done systematically, and optionally by the hooks for long-lived/blocking logic.

Reply:
For this PR, I'll try and avoid behaviour changes and in a subsequent PR, we can look at cleaning this up.